library/bignum.c

```diff
@@ -1394,44 +1394,42 @@ void mpi_mul_hlp( size_t i,
 {
     mbedtls_mpi_uint c = 0, t = 0;
 
-#if defined(MULADDC_HUIT)
-    for( ; i >= 8; i -= 8 )
-    {
-        MULADDC_INIT
-        MULADDC_HUIT
-        MULADDC_STOP
-    }
+    /*
+     * Unroll 8 times; this provides a reasonable compromise across platforms.
+     *
+     * Unrolling less hurts performance of FFDH/RSA on some platforms (for
+     * example, unrolling 4 rather than 8 times decreases perfomance by around
+     * 12% on Cortex-M4 cores). Unrolling more increase the code size linearly
+     * (for example, unrolling 16 rather than 8 times would increase the code
+     * size by around 250 bytes on Cortex-M0).
+     *
+     * Also, on 32-bit platforms, 256-bit numbers are 8 limbs, and this is a
+     * common size for ECC, widely used on constrained platforms.
+     *
+     * Use optimized 8/4/2-times version if available.
+     */
+#if defined(MULADDC_FAST2)
```

Comment:
Nit: Naming-wise I would prefer to stick with `MULADDC_X` instead of `MULADDC_FASTX`, and have 
```c
#if !defined(MULADDC_2)
#define MULADDC_2 MULADDC_CORE MULADDC_CORE
#endif  
```
at the bottom of `bn_mul.h` where all those macros are defined.

Reply:
The idea with using a different name was that `mul_hlp` might want to make different unrolling decisions depending on whether the 8-times version is optimized or not, but it turned out not to be necessary, so indeed it's simpler to have a single name.